src/mscorlib/corefx/System/Globalization/StringInfo.cs

```diff
@@ -119,7 +119,61 @@ public int LengthInTextElements
             }
         }
 
-        public static String GetNextTextElement(String str)
+        public string SubstringByTextElements(int startingTextElement) 
+        {
+            // If the string is empty, no sense going further. 
+            if (null == this.Indexes) 
+            {
+                // Just decide which error to give depending on the param they gave us....
+                if (startingTextElement < 0) 
+                {
+                    throw new ArgumentOutOfRangeException("startingTextElement", SR.ArgumentOutOfRange_NeedPosNum);
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("startingTextElement", SR.Arg_ArgumentOutOfRangeException);
+                }
+            }
+            return (SubstringByTextElements(startingTextElement, Indexes.Length - startingTextElement));
+        }
+
+        public string SubstringByTextElements(int startingTextElement, int lengthInTextElements) 
+        {
+            if (startingTextElement < 0)
+            {
+                throw new ArgumentOutOfRangeException("startingTextElement", SR.ArgumentOutOfRange_NeedPosNum);
```

Comment:
We have not done the nameof reformatting in CoreCLR and CoreRT yet; and our coding guidelines say to prefer consistency with the existing style...